### PR TITLE
Stack Exchange authentication via OAuth 2.0

### PR DESCRIPTION
**Pull request by MichalFoksa**
Plus support response in gzip (Content-Encoding: gzip).

### CODE REVIEW

**Comment by kullfar** [comment]
I have cherry-picked your commit with gzip support to the master branch (with small fixes).
Rebase please this Pull request and provide me test user and test application credential, so I can run the example by myself

**Comment by kullfar** [comment]
Thanks!

File: scribejava-apis/src/main/java/com/github/scribejava/apis/StackExchangeApi.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * @author Michal Foksa
+ *
+ */
+public class StackExchangeApi extends DefaultApi20 {
+
+    private static final String AUTHORIZE_URL
+    = "https://stackexchange.com/oauth?client_id=%s&redirect_uri=%s";
+
+    private StackExchangeApi() {
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/StackExchangeApi.java]: make it protected. It was my fault to make all API constructors private. I fixed that. In master branch all API constructors are protected, so anyone can extend API in their code if they want. But couldn't instantiate directly, forcing them to use singleton

**Comment by kullfar** [comment]
the rest of code is ok (just make protected constructor please)

**Comment by MichalFoksa** [comment]
Here you are, protected constructor.

**Comment by MichalFoksa** [comment]
I can create a temporary test application, but I cannot create a test account there.

**Comment by kullfar** [comment]
why not?

**Comment by kullfar** [comment]
You can create test email and register new user anywhere, right?

**Comment by MichalFoksa** [comment]
I would delete test app after the PR merge, but I cannot delete an account, and creating single purpose account is nasty. Actually having an account on `stackoverflow.com` might be beneficial for you in future - if you have to test the `StackExchangeExample.java`.

**Comment by kullfar** [comment]
It's not secure to give permissions to access my real account for an unknown test application ;-)
FB have such feature as special temporary test account for a application. Nice decision I think.
If stackoverflow doesn't have such feature, we have no chance to not create single purpose temporary account ;-(

**Comment by MichalFoksa** [comment]
An email address to send you test data, PLS.

**Comment by kullfar** [comment]
Use this one pls
https://github.com/scribejava/scribejava/blob/master/pom.xml#L48

**Comment by kullfar** [comment]
I have merged this Pull Request via console (not by github.com). So closing it here

**Pull request closed without merging**